src/components/Layout/PhoneNumberInput.tsx

```diff
@@ -77,7 +87,7 @@ export const PhoneNumberInput: FunctionComponent<{
         <TextInput
           style={styles.numberInput}
           keyboardType="phone-pad"
-          value={mobileNumberValue}
+          value={value}
```

Comment:
Can we approach this differently? Right now the `value` corresponds to 2 things: what the user sees and also what is used as the underlying state. These 2 things can actually be separate.

So the underlying state remains as what's passed in to this component (`mobileNumberValue`), but when we pass it over to `TextInput` for rendering, we transform it first,  like what you did. When the user changes the text in the input box, `onChange`  will receive the text and transform it back to the original type. In this case,  a number without the spaces.

This illustrates one key thing about frontend rendering btw, what is rendered depends on the state, and these are likely of different types. Another example is when we want to render money. We can store it in cents, but present it   in a  nice human readable way with  the dollar sign and commas. When the user edits it, we will  transform what the user has edited back to cents for storing in our state.

Reply:
Also,  by doing it this way, consumers of `PhoneNumberInput` don't need to know about the underlying state in it. The consumers will  just know that this renders a  phone number nicely and when a  user edits it, the  value will be feedback to the consumer as a string without spaces.